### PR TITLE
SL-192: new field for customizing the description parameter

### PR DESCRIPTION
**Pull request by mant02**
Reused Gytautas implementation but updated with other changes to avoid conflicts ;D

Added a setting for custom text that is used for the Description parameter when sending emails.

### CODE REVIEW

File: controllers/admin/AdminSaferPayOfficialSettingsController.php [hunk 1]
```diff
@@ -381,6 +384,32 @@ private function getFieldOptionsOrderState()
...
+            'fields' => [
+                SaferPayConfig::SAFERPAY_PAYMENT_DESCRIPTION => [
+                    'title' => $this->module->l('Description', self::FILE_NAME),
+                    'type' => 'text',
+                    'desc' => 'This description is visible in payment page also in payment confirmation email',
+                    'class' => 'fixed-width-xxl'
+                ],
+            ],
+            'buttons' => [
+                'save_and_connect' => [
```
Comment (GytisZum) [inline controllers/admin/AdminSaferPayOfficialSettingsController.php:403]: save and connect ? what this functionality connects ?

File: src/Config/SaferPayConfig.php [hunk 2]
```diff
@@ -241,10 +246,6 @@ class SaferPayConfig
     const TRANSACTION_STATUS_PENDING = 'PENDING';
     const TRANSACTION_STATUS_CANCELED = 'CANCELED';
 
-    const LOG_TYPE_SUCCESS = 'SUCCESS';
```
Comment (GytisZum) [inline src/Config/SaferPayConfig.php:244]: Why these was deleted ?

Comment (mant02) [reply src/Config/SaferPayConfig.php:244]: they weren't used anywhere

File: src/Service/Request/RequestObjectCreator.php [hunk 3]
```diff
@@ -118,6 +118,7 @@ public function createPayment(Cart $cart, $totalPrice)
         $payment->setValue($totalPrice);
         $payment->setCurrencyCode($currency['iso_code']);
         $payment->setOrderReference($order->reference);
+        $payment->setDescription(Configuration::get(SaferPayConfig::SAFERPAY_PAYMENT_DESCRIPTION));
```
Comment (mandan2) [inline src/Service/Request/RequestObjectCreator.php:121]: cast to string

**Review by mandan2 — APPROVED** [review]
one comment, if it works then good

**Pull request merged**